02-microarray/dimension-reduction_microarray_01_pca.Rmd

```diff
@@ -0,0 +1,307 @@
+
+## Perform Principal Components Analysis
+
+In this code chunk, we are going to perform Principal Component Analysis (PCA) on our data and create a data frame using the PCA scores and the variables from our metadata that we are going to use to annotate our plot later.
+We are using the base R `prcomp()` function to perform Principal Component Analysis (PCA) here [@prcomp-function].
+
+```{r}
+# Perform Principal Component Analysis (PCA)
+pca <- prcomp(t(df), # We have to transpose our data frame to meet the `prcomp()` function requirements
+  scale = TRUE # This tells R that we want the variables scaled to have unit variance
+)
+
+# Make the first 10 principal components into a data frame for plotting with `ggplot2`
+pca_df <- data.frame(pca$x[, 1:10]) %>%
+  # Turn samples_ids stored as rownames into column
+  tibble::rownames_to_column("refinebio_accession_code") %>%
+  # Bring only the variables that we want from the metadata into this data frame -- here we are going to join by `refinebio_accession_code` values
+  dplyr::inner_join(dplyr::select(
+    metadata, refinebio_accession_code,
+    histology, subgroup
+  ),
+  by = "refinebio_accession_code"
+  )
+```
+
+Before visualizing and interpreting the results, it can be useful to understand the proportion of variance explained by each principal component.
+For example, if PC1 explained 96% of the variance in your data and very clearly showed a difference between sample batches you would be very concerned about your dataset!
+On the other hand, if a separation of batches was apparent in a different principal component that explained a low proportion of variance and the first few PCs explained most of the variance and appeared to correspond to something like tissue type and treatment, you would be less concerned [@CCDL2020-cluster-validation].
+
+The `summary()` function reports the proportion of variance explained by each principal component.
+By accessing the `importance` element, which contains the proportion of variance explained by each principal component, with `<summary results>$importance`, we can use indexing to only look at the first `n` PCs.
+
+```{r}
+# Save the summary of the PCA results using the `summary()` function
+pca_summary <- summary(pca)
+
+# Now access and store the importance information for the first 10 PCs
+pca_importance <- pca_summary$importance[, 1:10]
+
+# Let's take a look at the importance information
+pca_importance
+```
+
+Now that we've seen the proportion of variance for the first ten PCs, let's plot the PC scores for the first two principal components, the components responsible for the most explained proportion of variance in our dataset. 
+
+Let's also label the data points based on their genotype subgroup since this is central to the subgroup specific based hypothesis in the [original paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3683624/#SD1) [@Northcott2012].
```

Comment:
```suggestion
Let's also label the data points based on their genotype subgroup.
```
I don't know if we need to use the original paper as a rationale for what we are doing now, but I am for giving a rationale for why we use `subgroup`.